File: .NET Framework 4 Base64 Converter/[C#][VB]-.NET Framework 4 Base64 Converter/C# and VB/C#,VB/Base64ConverterCS/Switches.cs
using System;
using System.Text;

internal class Switches
{
   internal string Switch;

   public Switches(string switch1)
   {
      this.Switch = switch1;
   }

   public static int GetBufferSize(string[] args)
   {
      foreach (var arg in args) {
         if (arg.Trim().ToUpper().StartsWith("/S:")) {
            int bufferSize;
            var result = arg.Remove(0, 3);
            if (Int32.TryParse(result, out bufferSize))
               return bufferSize;
            break;
         }
      }
      return Encode64.DefaultSize;
   }

   public bool IsSwitchSet(string value)
   {
      return value.ToUpper().Equals(this.Switch.ToUpper());
   }

   public bool HasSwitch(string value)
   {
      return value.Trim().ToUpper().StartsWith(this.Switch.ToUpper());
   }
}

File: .NET Framework 4 Base64 Converter/[C#][VB]-.NET Framework 4 Base64 Converter/C# and VB/C#,VB/Base64ConverterVB/Switches.vb
Friend Class Switches
   Friend Switch As String

   Public Sub New(ByVal switch As String)
      Me.Switch = switch
   End Sub

   Public Shared Function GetBufferSize(ByVal args() As String) As Integer
      For Each arg In args
         If arg.Trim().ToUpper().StartsWith("/S:") Then
            Dim bufferSize As Integer
            arg = arg.Remove(0, 3)
            If Int32.TryParse(arg, bufferSize) Then
               Return bufferSize
            End If
            Exit For
         End If
      Next
      Return Encode64.DefaultSize
   End Function

   Public Function IsSwitchSet(ByVal value As String) As Boolean
      Return value.ToUpper().Equals(Me.Switch.ToUpper())
   End Function

   Public Function HasSwitch(ByVal value As String) As Boolean
      Return value.Trim().ToUpper().StartsWith(Me.Switch.ToUpper())
   End Function
End Class